### PR TITLE
Improve listing of available targets

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Platform option is now an enum.
- Add new exit code with platform validation.
- Rework code accordingly.
- Rework target listing.
- Add new option listtargets to replace listboards.
- Improve output with helpful messages.

## Motivation and Context
- Improve usability and provide helpful messages whenever possible.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: nanoFirmwareFlasher/FirmwarePackage.cs [hunk 1]
```diff
@@ -64,23 +63,75 @@ internal abstract class FirmwarePackage : IDisposable
...
+
+                    case SupportedPlatform.stm32:
+                        query.Add("ST");
+                        query.Add("ORGPAL");
+                        query.Add("NETDUINO3");
+                        query.Add("QUAIL");
+                        query.Add("GHI");
+                        query.Add("IngenuityMicro");
+                        query.Add("WeAct");
+                        break;
```
Comment (Ellerbach) [inline nanoFirmwareFlasher/FirmwarePackage.cs:105]: more generally, is there a way to add the platform in the CloudSmith metadata?

File: nanoFirmwareFlasher/FirmwarePackage.cs [hunk 2]
```diff
@@ -64,23 +63,75 @@ internal abstract class FirmwarePackage : IDisposable
...
-                if( string.IsNullOrEmpty(filter))
+                List<string> query = new(); 
+
+                switch(platform)
                 {
-                    Console.Write(" without filter");
+                    case SupportedPlatform.esp32:
+                        query.Add("ESP");
+                        query.Add("M5");
+                        query.Add("Pyb");
```
Comment (Ellerbach) [inline nanoFirmwareFlasher/FirmwarePackage.cs:92]: it's STM :-)

File: nanoFirmwareFlasher/FirmwarePackage.cs [hunk 3]
```diff
@@ -64,23 +63,75 @@ internal abstract class FirmwarePackage : IDisposable
...
+                        query.Add("FEATHER");
```
Comment (Ellerbach) [inline nanoFirmwareFlasher/FirmwarePackage.cs:93]: missing lilygo

**Review by Ellerbach — COMMENTED** [review]
a little missmatch in the platform name. Other than that, all good

File: nanoFirmwareFlasher/FirmwarePackage.cs (hunk 1 above)
Comment (josesimoes) [reply nanoFirmwareFlasher/FirmwarePackage.cs:105]: TDB!

File: nanoFirmwareFlasher/FirmwarePackage.cs (hunk 2 above)
Comment (josesimoes) [reply nanoFirmwareFlasher/FirmwarePackage.cs]: 🤦🏻‍♂️ of course!!

File: nanoFirmwareFlasher/FirmwarePackage.cs (hunk 3 above)
Comment (josesimoes) [reply nanoFirmwareFlasher/FirmwarePackage.cs:92]: That will be picked up by the ESP filter, the target name is `ESP32_LILYGO`.

**Pull request merged**